src/util/importJSON.ts

```diff
@@ -25,6 +25,10 @@ import {
 interface ImportHtmlOptions {
   skipRoot? : boolean,
 }
+
+/** Insert the given value at the context. Modifies contextIndex and thoughtIndex. */
+type insertThought = (value: string, context: Context, rank: number, created?:Timestamp, lastUpdated?: Timestamp) => void
```

Comment:
Thanks for eliminating the duplicate code! 

Another way to do this is to type `insertThought` inline and then use `typeof insertThought` to type the `saveThoughts` parameter.